Library/Formula/hbase.rb

```diff
@@ -4,32 +4,148 @@ class Hbase < Formula
+      EOS
+
+    (var/"run/hbase").mkpath
+    (libexec/"logs").mkpath
   end
 
-  def caveats; <<-EOS.undent
-    You must edit the configs in:
-      #{libexec}/conf
-    to reflect your environment.
+  plist_options :manual => "#{HOMEBREW_PREFIX}/opt/hbase/bin/start-hbase.sh"
 
-    For more details:
-      https://hbase.apache.org/book.html
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      #{(build.without? "hadoop") ? "<true/>" : "<dict>\n        <key>OtherJobEnabled</key>\n        <string>"+Formula["hadoop"].plist_name+"</string>\n      </dict>"}
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>EnvironmentVariables</key>
+      <dict>
+       <key>HBASE_MASTER_OPTS</key><string> -XX:PermSize=128m -XX:MaxPermSize=128m</string>
+       <key>HBASE_LOG_DIR</key><string>#{var}/hbase</string>
+       <key>HBASE_HOME</key><string>#{opt_libexec}</string>
+       <key>HBASE_SECURITY_LOGGER</key><string>INFO,RFAS</string>
+       <key>HBASE_PID_DIR</key><string>#{var}/run/hbase</string>
+       <key>HBASE_NICENESS</key><string>0</string>
+       <key>HBASE_IDENT_STRING</key><string>root</string>
+       <key>HBASE_REGIONSERVER_OPTS</key><string> -XX:PermSize=128m -XX:MaxPermSize=128m</string>
+       <key>HBASE_OPTS</key><string>-XX:+UseConcMarkSweepGC</string>
+       <key>HBASE_ROOT_LOGGER</key><string>INFO,RFA</string>
+       <key>HBASE_LOG_PREFIX</key><string>hbase-root-master</string>
+       <key>HBASE_LOGFILE</key><string>hbase-root-master.log</string>
+      </dict>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/hbase</string>
+        <string>--config</string>
+        <string>#{opt_libexec}/conf</string>
+        <string>master</string>
+        <string>start</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/hbase/hbase.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/hbase/hbase.err</string>
+    </dict>
+    </plist>
     EOS
   end
 
   test do
     assert_match /#{version}/, shell_output("#{bin}/hbase mapredcp")
+
+    cp_r (libexec/"conf"), testpath
+    inreplace (testpath/"conf/hbase-site.xml") do |s|
+      s.gsub! /(hbase.rootdir.*)\n.*/, "\\1\n<value>file://#{testpath}/hbase</value>"
+      s.gsub! /(hbase.zookeeper.property.dataDir.*)\n.*/, "\\1\n<value>#{testpath}/zookeeper</value>"
+    end
+
+    ENV["HBASE_LOG_DIR"]  = (testpath/"logs")
+    ENV["HBASE_CONF_DIR"] = (testpath/"conf")
+    ENV["HBASE_PID_DIR"]  = (testpath/"pid")
+
+    system "#{bin}/start-hbase.sh"
+    sleep 2
+    assert_match /Zookeeper/, pipe_output("nc 127.0.0.1 2181 2>&1", "stats")
+    system "#{bin}/stop-hbase.sh"
```

Comment:
This needs to be in a `begin`/`ensure` block otherwise the `assert_match` will stop it being stopped.